### ISSUE TITLE
[QUESTION] reproducible steps that would yield unsaved changes from a prior closed session 

### ISSUE THREAD

**Issue by nonumeros** [mawww/kakoune#4861]
### Question

How could I possibly trigger some changes on a buffer from a session that was closed? 

In other words, the session was closed.  The session for kak that is, not the terminal session.

And Namely, `g-g` and `g-e` were used after opening the buffer with the same contents as before.

Could someone give me an idea as to if  the above would be at all possible. And if so. What would be an easy and reproducible method with steps to accomplish it? 

As you could probably guess from the desc above, I've been trying to reproduce it to no avail. 

But I'm sure there must a possible scenario that the above could be reproduced… If not from kak, from the terminal session then. 

Also. The terminal session was never closed. Suffice to say, it was never aborted nor suspended either. The buffer with kak contents were no longer there. I can only assume that the session for kakoune was closed (at least the particular session where the terminal session in question was running), but I must confess that I never checked for the number of kakoune processes that may have been running at the same time. 

Bottom line to reproduce. 

* Open the closed kak session in the same terminal session (still open)

* Use `g-g` or `g-e` on the buffer with the file contents

* While trying to close off that very same kak session, it would warn me about `unsaved changes` (But of course, no modifications had taken place. And I can almost assure you that no changes had taken place within that session)

* Had to «undo» the changes (even though no changes were made on that same buffer, other than the aforementioned operations such as `g-g` or `g-e` on that buffer

And…

I must clarify that before I merged (non-git related) the changes, the file in question was copied, so I could look up any supposedly further  modifications with say, `diff` between what kak was actually warning aboutt with with its `unsaved changes` and the contents of the file before it was opened with the aforementioned kakoune session again.

The results were sort of surprising 

The file had indeed been modified but the kak session had been closed off 

Running kak at fa060c2a

**Comment by nonumeros** [comment]
Mm… after checking the revision fa060c2  of which is what I'm running  and looking at the title of **Fix fatal exception when checking if buffer needs to be reloaded If, for example, the buffer path now is a directory** is not clear whether the path is an absolute or relative. Assuming it's a a relative path, the problem with the *unsaved changes* from a file that had not *reloaded* within an absolute path, as it should have been, it's nevertheless concerning.  

Hopefully this issue is only noise.

**Comment by nonumeros** [comment]
Going over it again as a reminder for even myself  about this behavior:

In order to close  the file and the kak session successfully, `u` had to be used or else it kept warning me about `unsaved changes` 

But neither `:edit` nor any kind of `:w[rite]` operation was ever used. 

Why would I edit it or write to the buffer if that was never the intention. 

Let's call the file `a.txt` and after this behavior I copied that very same  file `a.txt` to the same location (the location here is irrelevant really, since I wanted to get any notable `diff` between the supposed file that had `unsaved changes` , and the same file before it was opened, so I copied it to `a.bak.txt` 

This is more than definitive, there were changes *before* and *after* from the time that had lapsed when the file was opened and closed afterwards.
